kernel/src/hil/spi.rs

```diff
@@ -91,22 +92,22 @@ pub trait SpiMaster {
         write_buffer: &'static mut [u8],
         read_buffer: Option<&'static mut [u8]>,
         len: usize,
-    ) -> Result<(), ErrorCode>;
-    fn write_byte(&self, val: u8);
-    fn read_byte(&self) -> u8;
-    fn read_write_byte(&self, val: u8) -> u8;
+    ) -> Result<(), (ErrorCode, &'static mut [u8], Option<&'static mut [u8]>)>;
+    fn write_byte(&self, val: u8) -> Result<(), ErrorCode>;
+    fn read_byte(&self) -> Result<u8, ErrorCode>;
+    fn read_write_byte(&self, val: u8) -> Result<u8, ErrorCode>;
 
     /// Tell the SPI peripheral what to use as a chip select pin.
     /// The type of the argument is based on what makes sense for the
     /// peripheral when this trait is implemented.
-    fn specify_chip_select(&self, cs: Self::ChipSelect);
+    fn specify_chip_select(&self, cs: Self::ChipSelect) -> Result<(), ErrorCode>;
 
     /// Returns the actual rate set
-    fn set_rate(&self, rate: u32) -> u32;
+    fn set_rate(&self, rate: u32) -> Result<u32, ErrorCode>;
     fn get_rate(&self) -> u32;
-    fn set_clock(&self, polarity: ClockPolarity);
+    fn set_clock(&self, polarity: ClockPolarity) -> Result<(), ErrorCode>;
     fn get_clock(&self) -> ClockPolarity;
-    fn set_phase(&self, phase: ClockPhase);
+    fn set_phase(&self, phase: ClockPhase) -> Result<(), ErrorCode>;
```

Comment:
What if it's an I2C-to-SPI bridge? 

https://www.nxp.com/products/peripherals-and-logic/signal-chain/bridges/ic-bus-to-spi-bridge:SC18IS602B

But I agree with you Brad, I think there shouldn't be errors. This is a pretty tiny edge case, and we don't want to add code size/complexity everywhere for it. It would be fine to have a different SPI interface for an I2C-to-SPI bridge, for which a (software) bridge capsule could then filter out errors if you wanted to use the standard SPIMaster API.